### PR TITLE
Update NUnit

### PR DESCRIPTION
**Pull request by jskeet**
3 stages:

- Update the project files
- Fix a CultureSaver issue under netcoreapp2.0 with recent NUnit
- Fix a FakeDateTimeZoneSource bug and the test

### CODE REVIEW

File: src/NodaTime.Test/Testing/TimeZones/FakeDateTimeZoneSourceTest.cs [hunk 1]
```diff
@@ -129,15 +129,17 @@ public void GetEnumerator()
         // Unfortuntely NUnit requires the exact exception type :(
         private static void AssertBuildFails(FakeDateTimeZoneSource.Builder builder)
         {
+            bool success = false;
             try
             {
                 builder.Build();
-                Assert.Fail("Expected exception");
```
Comment (malcolmr) [inline src/NodaTime.Test/Testing/TimeZones/FakeDateTimeZoneSourceTest.cs]: Nice fix. But how is this related to the NUnit upgrade? (I don't mind either way, just curious.)

Comment (jskeet) [reply src/NodaTime.Test/Testing/TimeZones/FakeDateTimeZoneSourceTest.cs]: While we were still catching the exception in the latest NUnit, it was being reported as a failure - I suspect that there's some out-of-band way that it records errors, even if an exception isn't thrown.

Comment (jskeet) [reply src/NodaTime.Test/Testing/TimeZones/FakeDateTimeZoneSourceTest.cs]: (I haven't dug into it though - that's just a guess.)

Comment (malcolmr) [reply src/NodaTime.Test/Testing/TimeZones/FakeDateTimeZoneSourceTest.cs]: Doh, and you even said as much in the commit message, which I did read.

Thanks for explaining (again!)
